src/Store.js

```diff
@@ -302,49 +374,107 @@ class Store {
       })
     )
 
-    return query.then(recordsFromServer => recordsInStore.concat(...recordsFromServer))
+    return query.then((recordsFromServer) =>
+      recordsInStore.concat(...recordsFromServer)
+    )
   }
 
   /**
-   * finds all of the instances of a given type. If there are instances available in the store,
-   * it will return those, otherwise it will trigger a fetch
+   * Builds fetch url based
    *
-   *   store.findAll('todos')
-   *   // fetch triggered
-   *   => [event1, event2, event3]
-   *   store.findAll('todos')
-   *   // no fetch triggered
-   *   => [event1, event2, event3]
+   * @method fetchUrl
+   * @param {String} type the type to find
+   * @param {Object} options
+   */
+  fetchUrl (type, queryParams, id, options) {
+    const { baseUrl, modelTypeIndex } = this
+    const { endpoint } = modelTypeIndex[type]
+
+    return requestUrl(baseUrl, endpoint, queryParams, id, options)
+  }
+
+  /**
+   * Gets all records with the given `type` from the store. This will never fetch from the server.
    *
-   * passing `fromServer` as an option will always trigger a
-   * fetch if `true` and never trigger a fetch if `false`.
-   * Otherwise, it will trigger the default behavior
+   * @method getAll
+   * @param {String} type the type to find
+   * @param {Object} options
+   * @return {Array} array of records
+   */
+  getAll = (type, options = {}) => {
+    const { queryParams } = options
+    if (queryParams) {
+      return this.getCachedRecords(type, queryParams)
+    } else {
+      return this.getRecords(type)
+    }
+  }
+
+  /**
+   * Finds all records with the given `type`. Always fetches from the server.
    *
-   *   store.findAll('todos', { fromServer: false })
-   *   // no fetch triggered
-   *   => []
+   * @async
+   * @method fetchAll
+   * @param {String} type the type to find
+   * @param {Object} options
+   * @return {Promise} Promise.resolve(records) or Promise.reject(status)
+   */
+  async fetchAll (type, options = {}) {
+    const store = this
+    const { queryParams } = options
+    const url = this.fetchUrl(type, queryParams)
+    const response = await this.fetch(url, { method: 'GET' })
+
+    if (response.status === 200) {
+      this.data[type].cache.set(url, [])
+      const json = await response.json()
+
+      if (json.included) {
+        this.createModelsFromData(json.included)
+      }
+
+      return transaction(() =>
+        json.data.map((dataObject) => {
+          const { id, attributes = {}, relationships = {} } = dataObject
+          const ModelKlass = this.modelTypeIndex[type]
+          const record = new ModelKlass({
+            store,
+            relationships,
+            ...attributes
+          })
+          const cachedIds = this.data[type].cache.get(url)
+          this.data[type].cache.set(url, [...cachedIds, id])
+          this.data[type].records.set(String(id), record)
+          return record
+        })
+      )
```

Comment:
I'm having trouble understanding how this returns `Promise.resolve(records)` but that might be better explained via chat (unless you feel like typing it out isn't a huge hassle).

Reply:
Oh that was just a doc issue - I'll update the docs. We WILL be changing this to always return Promises (even if they don't fetch) but that's a bigger breaking change coming later.

Reply:
Actually no this does return a promise - I will get back to you on why

Reply:
This method was already here I think I just moved it btw - but async methods always return promises